src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,376 @@
+                  'iui-selected': isSamePeriod(value, selectedTime),
+                })}
+                key={value}
+                tabIndex={isSamePeriod(value, selectedTime) ? 0 : undefined}
+                // ref={(ref) => {
+                //   scrollIntoView(ref, isSamePeriod(value, selectedTime));
+                // }}
+                onClick={() => onPeriodClick(value)}
+              >
+                {value}
+              </li>
+            ))}
+          </ol>
+        </div>
+      )}
+    </>
+  );
+};
+
+export type TimePickerColumnProps = {
+  /**
+   * Data to render in column.
+   */
+  data: number[];
+  /**
+   * Current focused time.
+   */
+  focusedTime: Date;
+  /**
+   * Current selected time.
+   */
+  selectedTime?: Date;
+  /**
+   * Callback when focus is changed.
+   */
+  onFocusChange?: (value?: number) => void;
+  /**
+   * Callback when date is changed.
+   */
+  onSelectChange?: (value: number) => void;
+  /**
+   * Compare function.
+   */
+  isSame: (value: number, date: Date | undefined) => boolean;
+  /**
+   * Step.
+   */
+  step?: number;
+};
+
+const TimePickerColumn = (props: TimePickerColumnProps): JSX.Element => {
+  const {
+    data,
+    focusedTime,
+    selectedTime,
+    onFocusChange,
+    onSelectChange,
+    isSame,
+    step = 0,
+  } = props;
+  const needFocus = React.useRef(false);
+
+  React.useEffect(() => {
+    if (needFocus.current) {
+      needFocus.current = false;
+      // onFocusChange?.(undefined);
+    }
+  });
+
+  const scrollIntoView = (ref: HTMLLIElement | null, isSame: boolean) => {
+    isSame && ref?.scrollIntoView({ block: 'nearest', inline: 'nearest' });
+  };
+
+  const handleTimeKeyDown = (
+    event: React.KeyboardEvent<HTMLLIElement>,
+    maxValue: number,
+    onFocus: (value: number) => void,
+    onSelect: (value: number) => void,
+    currentValue?: number,
+  ) => {
+    if (currentValue == null) {
+      return;
+    }
+    switch (event.key) {
+      case 'ArrowDown':
+        if (currentValue + 1 > maxValue) {
+          break;
+        }
+        onFocus(currentValue + 1);
+        needFocus.current = true;
+        event.preventDefault();
+        break;
+      case 'ArrowUp':
+        if (currentValue - 1 < 0) {
+          break;
+        }
+        onFocus(currentValue - 1);
+        needFocus.current = true;
+        event.preventDefault();
+        break;
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        onSelect(currentValue);
+        event.preventDefault();
+        break;
+    }
+  };
+
+  return (
+    <div className='iui-time'>
+      <ol>
+        {data
+          .filter((_, index) => index % step === 0)
+          .map((value, index) => {
+            return (
+              <li
+                onKeyDown={(event) => {
+                  handleTimeKeyDown(
+                    event,
+                    data.length - 1,
+                    (index) => onFocusChange?.(data[index]),
```

Comment:
Are you going to export this to root index? If not then these functions should be required.

Reply:
changed